### PR TITLE
Code review: Testing the To-do list

### PR DESCRIPTION
**Pull request by Milypm**
For this project, we implement some testing methods.

The main categories are:
- Testing the logic for manipulating projects on localStorage
- Testing the logic for manipulating projects' tasks on localStorage
- Testing the DOM (event listeners)

Tools:
- Jest
- JavaScript (ES6)

### CODE REVIEW

File: __tests__/middleView.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  secondTask.click();
+  const taskTitle = document.getElementById('task-title-id');
+  const taskDescription = document.getElementById('task-description-id');
+  const taskDate = document.getElementById('task-date-id');
+  const taskPriority = document.getElementById('task-priority-id');
+  expect(taskTitle.textContent).toBe('Title: Finish project');
+  expect(taskDescription.textContent).toBe('Description: Work on last changes');
+  expect(taskDate.textContent).toBe('Due Date: 28 May 2021');
+  expect(taskPriority.textContent).toBe('Priority: Urgent');
+});
```
Comment (teekaytech) [inline __tests__/middleView.test.js]: **Required**

- [x] We must ensure that all test cases are passing. This particular one seems to be failing. Kindly check and fix the bug.

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've done really well so far. Here're the major highlights of this PR:__

- Your PR title, summary and commit messages are descriptive :+1: 
- You've created separate test files for each class or module :100: 
- Nice folder structure for the test files :ok_hand: 

__However, some changes that you are required to make before moving on to the next task. Please check my comments below and fix them as appropriate.__

### **Required Changes** :recycle: 

- [x] For this project, we must check if all functions (especially the logic-related ones) are tested. However, I believe you can write more tests that will cover some functions in `src/js/logics/projectsList-Store.js` and `src/js/logics/tasksList-Store.js` scripts.

- [x] Check the comments under the review.

### **Optional Changes** :yellow_circle: 
N/A


* * *
**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise.**

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@teekaytech) so I can see your comments and respond._

_**Please make the required changes and submit another review request when you are done.**_

* * *
Regards,
**Taofeek Olalere**
__Cheers and happy coding :smiley:__

**Review by umararshadbut — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @Milypm and @rloterh 
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

Kindly work on the suggested changes  👇.
- [x] There is one test case that is failing as the previous code reviewer suggested. Please ensure all the test cases are passing. 

##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**Umair**
**Happy Coding :smiley:**

File: __tests__/leftView.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import setLeftView from '../src/js/dom/leftView';
+import buildAll from '../src/js/build';
+
+const fs = require('fs');
```
Comment (rindrajosia) [inline __tests__/leftView.test.js]: [OPTIONAL] Please use import

File: __tests__/leftView.test.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+const path = require('path');
```
Comment (rindrajosia) [inline __tests__/leftView.test.js]: [OPTIONAL] Please use import

File: __tests__/middleView.test.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import setMiddleView from '../src/js/dom/middleView';
+import buildAll from '../src/js/build';
+
+const fs = require('fs');
```
Comment (rindrajosia) [inline __tests__/middleView.test.js]: [OPTIONAL] Please use import

File: __tests__/middleView.test.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+const path = require('path');
```
Comment (rindrajosia) [inline __tests__/middleView.test.js]: [OPTIONAL] Please use import

File: src/js/build.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import setLeftView from './dom/leftView';
+import setMiddleView from './dom/middleView';
+import setRightView from './dom/rightView';
+
+require('../css/style.css');
```
Comment (rindrajosia) [inline src/js/build.js:5]: [OPTIONAL] Please use import

File: src/js/logics/projectsList-Store.js [hunk 7]
```diff
@@ -53,7 +53,7 @@ const buildProject = (() => {
   };
 
   return {
-    projects, getProjects, addProject, editProject, deleteProject, findProject,
+    projects, projectObject, defaultProject, getProjects, addProject, editProject, deleteProject, findProject,
```
Comment (rindrajosia) [inline src/js/logics/projectsList-Store.js:56]: [OPTIONAL] Please add test units for: addProject, editProject, deleteProject **functions**

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- Good job by trying to fix bug related to test cases :+1: 

To improve :factory_worker: [REQUIRED]:

- [x] There is one test case that is failing as the previous code reviewer suggested. Please ensure all the test cases are passing.


Nice to have [OPTIONAL] :

- [x] Please use **import** instead of **require** as you are using ES6 syntax.
- [x] Make sure that you add test units for all functions in order to ensure that all of your functions work as expected.

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏

### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Pull request merged**